cmd/operator-sdk/test/local.go

```diff
@@ -220,10 +240,24 @@ func testLocalGoFunc(cmd *cobra.Command, args []string) error {
 	if tlConfig.goTestFlags != "" {
 		testArgs = append(testArgs, strings.Split(tlConfig.goTestFlags, " ")...)
 	}
-	if tlConfig.namespace != "" || tlConfig.noSetup {
-		testArgs = append(testArgs, "-"+test.SingleNamespaceFlag, "-parallel=1")
+	if tlConfig.operatorNamespace != "" || tlConfig.noSetup {
+		testArgs = append(testArgs, "-parallel=1")
 	}
-	env := append(os.Environ(), fmt.Sprintf("%v=%v", test.TestNamespaceEnv, tlConfig.namespace))
+	env := os.Environ()
+	if tlConfig.operatorNamespace != "" {
+		env = append(
+			env,
+			fmt.Sprintf("%v=%v", test.TestOperatorNamespaceEnv, tlConfig.operatorNamespace),
+		)
+	}
+
+	if cmd.Flags().Changed("watch-namespace") {
+		env = append(
```

Comment:
same here

Reply:
https://github.com/operator-framework/operator-sdk/pull/2617#discussion_r391679380

In addition if both `operatorNamespace` and `watchNamespace` is not set.
operatorNamespace is created during  while calling xtx.GetOperatorNamepspace() and watchNamespace is set to operator namespace.

I have considered all of those in units tests in 
pkg/test/resource_creator_test.go